napari/plugins/__init__.py

```diff
@@ -20,7 +37,122 @@
     plugin_manager.register(_builtins, name='builtins')
 
 
-__all__ = [
-    "PluginManager",
-    "plugin_manager",
-]
+dock_widgets: Dict[Tuple[str, str], Tuple[Type['QWidget'], dict]] = dict()
+function_widgets: Dict[Tuple[str, str], Tuple[Callable, dict, dict]] = dict()
+
+
+def register_dock_widget(
+    args: Union[DockWidgetArg, List[DockWidgetArg]],
+    hookimpl: HookImplementation,
+):
+    from qtpy.QtWidgets import QWidget
+
+    plugin_name = hookimpl.plugin_name
+    hook_name = '`napari_experimental_provide_dock_widget`'
+    for arg in args if isinstance(args, list) else [args]:
+        if isinstance(arg, tuple):
+            if not arg:
+                warn(
+                    f'Plugin {plugin_name} provided an invalid tuple to '
+                    f'{hook_name}.  Skipping'
+                )
+                continue
+            _cls = arg[0]
+            kwargs = arg[1] if len(arg) > 1 else {}
+        else:
+            _cls, kwargs = (arg, {})
+
+        if not isclass(_cls) and issubclass(_cls, QWidget):
+            warn(
+                f'Plugin {plugin_name} provided an invalid '
+                f'widget type to {hook_name}: {_cls!r}. Widget ignored.'
+            )
+            continue
+
+        if not isinstance(kwargs, dict):
+            warn(
+                f'Plugin {plugin_name} provided invalid kwargs '
+                f'to {hook_name} for class {_cls.__name__}. Widget ignored.'
+            )
+            continue
+
+        # Get widget name
+        name = str(kwargs.get('name')) or camel_to_spaces(_cls.__name__)
```

Comment:
```suggestion
        name = kwargs.get('name') or camel_to_spaces(_cls.__name__)
```
I think the `str` prevents the second thing from getting called. Was there a reason it was needed?

Reply:
ah right.  well, if for some reason they provide a non-string name, then it might cause an error when we use it as a menu item.  should use `str(kwargs.get('name', ''))`,  will change

Reply:
thanks!! can you change in `register_function_widget` too